predicate.go

```diff
@@ -0,0 +1,219 @@
+		}
+		for _, er := range erList.Items {
+			nodeAffinity := er.Spec.NodeAffinity
+			requirements := erc.Spec.MetadataRequirements
+
+			if nodeAffinity == nil ||
+				!mapInMap(requirements.MatchLabels, er.Spec.Properties) ||
+				!labelMatchesLabelSelectorExpressions(requirements.MatchExpressions, er.Spec.Properties) {
+				continue
+			}
+
+			nodeSelectorTerms := nodeAffinity.Required.NodeSelectorTerms
+			for _, node := range nodes {
+				// set er related with erc
+				if nodeMatchesNodeSelectorTerms(&node, nodeSelectorTerms) {
+					erc.Spec.ExtendedResourceNames = append(erc.Spec.ExtendedResourceNames, er.Name)
+					er.Spec.ExtendedResourceClaimName = erc.Name
+					extendedResources.Items = append(extendedResources.Items, er)
+					canSchedule = append(canSchedule, node)
+				}
+				canNotSchedule[node.ObjectMeta.Name] = "node's label is not satisfy er's nodeAffinity"
+			}
+		}
+
+		extendedResourceNum := erc.Spec.ExtendedResourceNum
+		if int64(len(canSchedule)) < extendedResourceNum {
+			canSchedule = canSchedule[:extendedResourceNum]
+			result.FailedNodes = canNotSchedule
+			glog.Error("The extendedresource that can be scheduled are insufficient")
+			result.Error = "The extendedresource that can be scheduled are insufficient"
+			return result
+		}
+
+		extenedResourceScheduler.UpdateExtendedResourceClaim(pod.Namespace, erc)
+		for _, er := range extendedResources.Items {
+			if er.Spec.ExtendedResourceClaimName != "" {
+				extenedResourceScheduler.UpdateExtendedResource(&er)
+			}
+		}
+		canSchedule = canSchedule[:extendedResourceNum]
+		result.FailedNodes = canNotSchedule
+	}
+
+	result.Nodes.Items = canSchedule
+	return result
+}
+
+func findExtendedResourceClaims(pod v1.Pod, extenedResourceScheduler *ExtendedResourceScheduler) ([]*v1alpha1.ExtendedResourceClaim, error) {
+	extendedResourceClaimNames := make([]string, 0)
+	for _, container := range pod.Spec.Containers {
+		if len(container.ExtendedResourceClaims) != 0 {
+			extendedResourceClaimNames = append(extendedResourceClaimNames, container.ExtendedResourceClaims...)
+		}
+	}
+	if len(extendedResourceClaimNames) == 0 {
+		return nil, errors.New("extendedresource not found")
+	}
+	extendedResourceClaims := make([]*v1alpha1.ExtendedResourceClaim, 0)
+	for _, ercName := range extendedResourceClaimNames {
+		erc, err := extenedResourceScheduler.FindExtendedResourceClaim(pod.Namespace, ercName)
+		if err != nil {
+			return nil, err
+		}
+		extendedResourceClaims = append(extendedResourceClaims, erc)
+	}
+	return extendedResourceClaims, nil
+}
+
+func findExtendedResourceList(erc *v1alpha1.ExtendedResourceClaim, extenedResourceScheduler *ExtendedResourceScheduler) (*v1alpha1.ExtendedResourceList, error) {
+	rawResourceName := erc.Spec.RawResourceName
+	erList, err := extenedResourceScheduler.FindERByRawResourceName(rawResourceName)
+	if err != nil {
+		return nil, err
+	}
+	extendedResourceNames := erc.Spec.ExtendedResourceNames
+	if len(extendedResourceNames) != 0 {
```

Comment:
you have misunderstood what i said. 
If ExtendedResourceNames in erc spec is set, it means users know exactly which ERs they want to consume. What we need to do is to check if the ERs exist and if the length of that ER array equals to ERNum in erc spec